File: OpenCV/test/C/DisplayImage.cpp
/*==============================================*/
/* Auther:   Willy Xie                          */
/* Filename: DisplayImage.cpp                   */
/* Version:  1.0                                */
/*==============================================*/
/* Description:                                 */
/* An example of using opencv to read and       */
/* display image.                               */
/*==============================================*/

// Reference : https://blog.gtwang.org/programming/ubuntu-linux-install-opencv-cpp-python-hello-world-tutorial/

#include <stdio.h>
#include <opencv2/opencv.hpp>

using namespace cv;

int main(int argc, char* argv[]) {
  // Check for existence of the input image
  if ( argc != 2 ) {
    printf("usage: DisplayImage.out <Image_Path>n");
    return -1;
  }

  // Read the frame
  Mat image;
  image = imread( argv[1], 1 );

  // Check whether the image is loaded properly
  if ( !image.data ) {
    printf("No image data n");
    return -1;
  }

  // Display the image until any keyboard interrupt is received
  namedWindow("Display Image", WINDOW_AUTOSIZE);
  imshow("Display Image", image);
  waitKey(0);

  return 0;
}

File: OpenCV/test/Python/display_image.py
#================================================#
#  Auther:   Willy Xie                           #
#  Filename: display_image.py                    #
#  Version:  1.0                                 #
#=============================================== #
#  Description:                                  #
#  An example of using opencv to read and        #
#  display image.                                #
#================================================#

# Reference : https://blog.gtwang.org/programming/ubuntu-linux-install-opencv-cpp-python-hello-world-tutorial/
#-*- coding: utf-8 -*-
import cv2
import sys

# Check for existence of the input image
if (len(sys.argv) != 2):
  print("usage: " + sys.argv[0] + " <Image_Path>n")
  sys.exit(-1);

# Read the frame
img = cv2.imread(sys.argv[1], cv2.IMREAD_COLOR)

# Display the image until any keyboard interrupt is received
cv2.namedWindow("Display Image", cv2.WINDOW_AUTOSIZE)
cv2.imshow('Display Image', img)
cv2.waitKey(0)
